tests/tokens_service_test.go

```diff
@@ -0,0 +1,101 @@
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/uuid"
+
+	"github.com/flow-hydraulics/flow-wallet-api/jobs"
+	"github.com/flow-hydraulics/flow-wallet-api/tests/internal/test"
+	"github.com/flow-hydraulics/flow-wallet-api/transactions"
+)
+
+func Test_TokensSetup(t *testing.T) {
+	cfg := test.LoadConfig(t, testConfigPath)
+	svc := test.GetServices(t, cfg).GetTokens()
+
+	_, testAccount, err := test.GetServices(t, cfg).GetAccounts().Create(context.Background(), true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type input struct {
+		sync      bool
+		tokenName string
+		address   string
+	}
+	type expect struct {
+		job   *jobs.Job
+		tx    *transactions.Transaction
+		error bool
+	}
+	testCases := []struct {
+		name   string
+		input  input
+		expect expect
+	}{
+		{
+			name: "success",
+			input: input{
+				sync:      false,
+				tokenName: "flowToken",
+				address:   testAccount.Address,
+			},
+			expect: expect{
+				job: &jobs.Job{
+					Type:                   transactions.TransactionJobType,
+					State:                  jobs.Init,
+					Error:                  "",
+					Result:                 "",
+					ExecCount:              int(1),
+					ShouldSendNotification: true,
+				},
+				tx: &transactions.Transaction{
+					TransactionType: transactions.FtSetup,
+					ProposerAddress: testAccount.Address,
+				},
+				error: false,
+			},
+		},
+		{
+			name: "fail address not found",
+			input: input{
+				sync:      false,
+				tokenName: "flowToken",
+				address:   "0x0ae53cb6e3f42a79",
+			},
+			expect: expect{
+				job:   nil,
+				tx:    nil,
+				error: true,
+			},
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			job, tx, err := svc.Setup(context.Background(), tc.input.sync, tc.input.tokenName, tc.input.address)
+			jobOpts := []cmp.Option{
```

Comment:
Currently this is undeterministic since the job may or may not be executed when comparing, I suggest waiting for the job to complete, for example:

```golang
// Wait for job to execute
for job != nil && job.State != jobs.Complete && job.State != jobs.Failed {
  time.Sleep(10 * time.Millisecond)
}
```